pkg/operator/controllers/machineset/machineset_controller.go

```diff
@@ -88,11 +102,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		modifiedMachineset.Spec.Replicas = to.Int32Ptr(int32(minSupportedReplicas-replicaCount) + *modifiedMachineset.Spec.Replicas)
 		err := r.client.Update(ctx, modifiedMachineset)
 		if err != nil {
+			r.log.Error(err)
+			conditions.SetControllerDegraded(ctx, r.client, cnds, err)
+
 			return reconcile.Result{}, err
 		}
 	}
 
-	return reconcile.Result{}, nil
+	return reconcile.Result{}, conditions.ClearControllerConditions(ctx, r.client, cnds)
```

Comment:
This confused me a bit, since at first glance it looks like the reconciler is now going to return some blank conditions as an error when there should be no error, but looking deeper I see that we're only returning the result from conditions.ClearControllerConditions as an error, which should be nil.

Do you have thoughts on setting/running the conditions before this final return? Not really necessary, but might improve readability.

Reply:
I think either way is fine, as long as we remain consistent. I'm happy to change this PR either way to establish that standard for future work. 

Perhaps "returning" the result of setting conditions might make more sense if that function was also responsible for crafting the return output? (e.g. make setting conditions a responsibility from exiting the function) I'm not immediately in favor of doing that, as it would conflate two separate responsibilities together, but if it's something we are consistent with it might make sense across the whole operator.